src/sentry/integrations/slack/webhooks/action.py

```diff
@@ -379,19 +384,22 @@ def open_resolve_dialog(self, slack_request: SlackActionRequest, group: Group) -
         #
         # [1]: https://stackoverflow.com/questions/46629852/update-a-bot-message-after-responding-to-a-slack-dialog#comment80795670_46629852
         org = group.project.organization
-        callback_id = {
+        callback_id_dict = {
             "issue": group.id,
             "orig_response_url": slack_request.data["response_url"],
             "is_message": _is_message(slack_request.data),
-            "tags": list(slack_request.get_tags()),
         }
         if slack_request.data.get("channel"):
-            callback_id["channel_id"] = slack_request.data["channel"]["id"]
-            callback_id["rule"] = slack_request.callback_data.get("rule")
-        callback_id = orjson.dumps(callback_id).decode()
+            callback_id_dict["channel_id"] = slack_request.data["channel"]["id"]
+            callback_id_dict["rule"] = slack_request.callback_data.get("rule")
+        callback_id = orjson.dumps(callback_id_dict).decode()
+
+        # only add tags to metadata
+        callback_id_dict["tags"] = list(slack_request.get_tags())
```

Comment:
not: for code clarity I suggest we copy callback_id_dict in metadata dict and then add tags to it instead of appending them to callback_id_dict.